cgo_cmd/vsnap_copy/snapshot_manager.go

```diff
@@ -57,3 +47,57 @@ func NewSnapshotManager(config *VSphereCreds) (*SnapshotManager, error) {
 		ivdPETM: ivdPETM,
 	}, nil
 }
+
+type VSphereCreds struct {
+	Host      string `json:"vchost"`
+	User      string `json:"vcuser"`
+	Pass      string `json:"vcpass"`
+	S3UrlBase string `json:"s3urlbase"`
+}
+
+func (v *VSphereCreds) Unmarshal(creds []byte) error {
+	return json.Unmarshal(creds, v)
+}
+
+func (v *VSphereCreds) Validate() error {
+	if v.Host == "" {
+		return fmt.Errorf("missing endpoint value")
+	}
+	if v.User == "" {
+		return fmt.Errorf("missing username value")
+	}
+	if v.Pass == "" {
+		return fmt.Errorf("missing password value")
+	}
+	if v.S3UrlBase == "" {
+		return fmt.Errorf("missing s3URLBase value")
+	}
```

Comment:
These error messages should refer to explicit field names or just bundle all checks into a single if statement and return a more ambiguous "invalid credential" message.  We don't really need such detailed errors.